pkg/webhook/webhook.go

```diff
@@ -99,6 +99,10 @@ func (m *podMutator) Handle(ctx context.Context, req admission.Request) (respons
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
+	podName := pod.GetName()
+	if podName == "" {
+		podName = pod.GetGenerateName()
```

Comment:
To make it clear that this is not the final name of the pod:

```suggestion
		podName = pod.GetGenerateName() + " (prefix)"
```